spec/libraries/acts_as_conditional_spec.rb

```diff
@@ -26,5 +26,30 @@
         expect(subject).to all be_instance_of(Course::Condition::Achievement)
       end
     end
+
+    describe('#conditions_satisfied_by?') do
+      subject { achievement }
+      let(:satisfied_condition) do
+        condition = double
```

Comment:
`double` is an empty object that I can stub 1 or 2 methods necessary for my test. I use it to avoid "lengthy initialization" e.g. to create a CourseUser, I need to write the tests within `with_tenant(:instance) do`. After using the factory method to create a `course_user`, I still need to use Rspec `allow` to stub the user behavior.